File: 228. Summary Ranges/228. Summary Ranges.java
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

class Solution {

  public List<String> summaryRanges(int[] nums) {
    if (nums.length == 0) {
      return Collections.emptyList();
    }

    List<String> summaryRanges = new ArrayList<>();
    int start = 0;

    for (int i = 1; i < nums.length + 1; i++) {
      if (i == nums.length || nums[i] != nums[i - 1] + 1) {
        if (start == i - 1) {
          summaryRanges.add(String.format("%d", nums[start]));
        } else {
          summaryRanges.add(String.format("%d->%d", nums[start], nums[i - 1]));
        }

        start = i;
      }
    }

    return summaryRanges;
  }
}

File: 228. Summary Ranges/228. Summary Ranges.py
from typing import List


class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if len(nums) == 0:
            return []

        summary_ranges = []
        start = 0

        for i in range(1, len(nums) + 1):
            if i == len(nums) or nums[i] != nums[i - 1] + 1:
                if start == i - 1:
                    summary_ranges.append(f"{nums[start]}")
                else:
                    summary_ranges.append(f"{nums[start]}->{nums[i - 1]}")
                start = i

        return summary_ranges
